demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\Statement\AssessmentTableExporter;
+
+use demosplan\DemosPlanCoreBundle\Entity\File;
+use Exception;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+
+class AssessmentTableZipExporter extends AssessmentTableXlsExporter
+{
+    protected array $supportedTypes = ['zip'];
+
+    /**
+     * @throws Exception
+     */
+    public function __invoke(array $parameters): array
+    {
+        $xlsxArray = parent::__invoke($parameters);
+
+        $statementAttachments = $this->getAttachmentsOfStatements($parameters['items']);
+
+        /** @var Xlsx $xlsxWriter */
+        $xlsxWriter = $xlsxArray['writer'];
+        $xlsxArray['writer'] = $this->writeReferencesIntoXlsx($xlsxWriter, $statementAttachments);
+
+        return [
+            'zipFileName' => $this->translator->trans('evaluation.assessment.table.export'),
+            'xlsx'        => $xlsxArray,
+            'attachments' => $statementAttachments,
+        ];
+    }
+
+    /**
+     * @return array<int, array<int, File>>
+     *
+     * @throws Exception
+     */
+    private function getAttachmentsOfStatements(array $statementIds): array
+    {
+        $files = [];
+        $index = 0;
+        foreach ($statementIds as $statementId) {
+            $statementAttachments =
+                $this->assessmentHandler->getStatementService()->getFileContainersForStatement($statementId);
+            $files[$index] = [];
+            foreach ($statementAttachments as $statementAttachment) {
+                $files[$index][] = $statementAttachment->getFile();
+            }
+            ++$index;
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param array<int, array<int, File> $files
+     */
+    private function writeReferencesIntoXlsx(Xlsx $xlsxWriter, array $files): Xlsx
+    {
+        $spreadsheet = $xlsxWriter->getSpreadsheet();
+        $sheet = $spreadsheet->getSheetByName($this->translator->trans('considerationtable'));
+
+        if (null === $sheet) {
+            $this->logger->error('No worksheet in xlsx for zip export!', [$sheet]);
```

Comment:
Can we create one in this case? What if no sheet exist, can we go further? Or should an exception or the like be thrown?

Reply:
I think in this case we should throw an exception. adjusted in [ad3134f](https://github.com/demos-europe/demosplan-core/pull/2337/commits/ad3134fe2f7f3d0f6509bd091b976174834e2064)

Reply:
thx. How is this caught and handled?

Reply:
I adjusted it a bit more: [3e35f11](https://github.com/demos-europe/demosplan-core/pull/2337/commits/3e35f1155dc80c3ccefb7901eb1f581f47a9258b)
It will be handled in the controller.